llvm/include/llvm/CAS/LazyMappedFileRegion.h

```diff
@@ -106,27 +112,36 @@ class LazyMappedFileRegion {
 
 private:
   Error extendSizeImpl(uint64_t MinSize);
-  void destroyImpl() {
-    if (FD) {
-      sys::fs::closeFile(*FD);
-      FD = None;
-    }
-  }
+  void destroyImpl();
   void moveImpl(LazyMappedFileRegion &RHS) {
     Path = std::move(RHS.Path);
     FD = std::move(RHS.FD);
     RHS.FD = None;
+#ifdef _WIN32
+    VM = RHS.VM;
+    MaxSize = RHS.MaxSize;
+    MappedRegions = RHS.MappedRegions;
+#else
     Map = std::move(RHS.Map);
+#endif
     CachedSize = RHS.CachedSize.load();
     RHS.CachedSize = 0;
     MaxSizeIncrement = RHS.MaxSizeIncrement;
 
+#ifndef _WIN32
     assert(!RHS.Map && "Expected Optional(Optional&&) to clear RHS.Map");
```

Comment:
This could go in the `#else` above, which would also make it closer to the move it is talking about.